app/services/trust_pilot_parser.rb

```diff
@@ -0,0 +1,67 @@
+class TrustPilotParser
+  attr_reader :reviews
+
+  def initialize(data)
+    @reviews = data['reviews'] || []
+  end
+
+  def parse_reviews
+    reviews.map do |review|
+      ReviewParser.new(review).parse
+    end
+  end
+
+  class ReviewParser
+    ATTRIBUTES = %w[response_id score comment title language created_at updated_at company_reply
+                    is_verified number_of_likes status report_data compliance_labels
+                    consumer_name consumer_location email user_reference_id].freeze
+    attr_reader :data
+
+    def initialize(data)
+      @data = OpenStruct.new(data)
+    end
+
+    def parse
+      ATTRIBUTES.inject({}) { |acc, key| acc.merge(key => extract_data(key)) }
+    end
+
+    private
+
+    def extract_data(key)
+      attr_name = key.camelize(:lower)
+      if data.respond_to?(attr_name)
+        data.send(attr_name)
```

Comment:
in OpenStruct all this public, properties. maybe better to use check for case? Because `respond_to` returns `private, protected` methods too.

Reply:
respond_to? by default doesn't return private and protected methods

Reply:
oh yes, only with param `true`. Ok, you can keep it as now